### ISSUE TITLE
Sitemap LastMod Schema Error

### ISSUE THREAD

**Issue by SEOOES** [sybrew/the-seo-framework#432]
**Describe the bug**
When LastModified in swiched ON in the settings, the sitemap has an error when running a sitemap validator check.

"19:0: ERROR: Element '{http://www.sitemaps.org/schemas/sitemap/0.9}lastmod': '2015-12-22T05:31-01:00' is not a valid value of the union type '{http://www.sitemaps.org/schemas/sitemap/0.9}tLastmod'."

When turning the LastModified option OFF, the error goes away. 

We'd like to obviously keep the LastModified add-on, so am reporting the bug. 

**How to reproduce the bug**
1. Set up a sitemap
2. WP admin > SEO > Sitemapsettings > Meta Data > Select " Add <lastmod> to the sitemap?"
3. Preview and test the sitemap on a site like https://tools.seochat.com/tools/site-validator/

**Expected behavior**
All clear, no issues. 

We love the plugin and hope this adjustment can be made. Also Updated to the latest version of the plugin, tested it again and the same occurred. 

Thank you.

**Comment by sybrew** [comment]
Hello 😃  

Thanks for opening this issue! I'd like to walk you through what's happening.

https://www.sitemaps.org/schemas/sitemap/0.9/sitemap.xsd defines `lastMod` as either:
- xsd:date -> ISO 8601, 5.2.1 -> `[-]CCYY-MM-DD[Z|(+|-)hh[[:]mm]]`
- xsd:dateTime -> ISO 8601, 5.4 -> `[-]CCYY-MM-DDThh[[:]mm[[:]ss]][Z|(+|-)hh[[:]mm]]`

What's between square brackets is optional. The pipe (`|`) means "or".

Either valid format is available via the "General -> Timestamp" settings in The SEO Framework.
With that, in our sitemap:
- xsd:date is presented as `CCYY-MM-DD`
- xsd:datetime is presented as `CCYY-MM-DDThh:mm+hh:mm`

If you enter other sitemaps, like the ones from Google (`https://www.google.com/sitemap.xml`), you'll find that the tool is flagging those are incorrect as well&mdash;or even blatantly crashing,

In other words, the tool from seochat is invalid. Please use these tools instead:
- https://search.google.com/search-console/sitemaps
- https://www.bing.com/webmaster/configure/sitemaps/home?url= (this link doesn't really work without entering your URL)
- https://webmaster.yandex.com/tools/sitemap/?tab=url

After all, they are the ones who you'd want to feed the sitemap to. 🚀
